lib/workers/repository/update/pr/changelog/bitbucket/index.spec.ts

```diff
@@ -0,0 +1,174 @@
+import * as _hostRules from '../../../../../../util/host-rules';
+import type { BranchUpgradeConfig } from '../../../../../types';
+import { getReleaseNotesMd } from '../release-notes';
+import { BitbucketChangeLogSource } from './source';
+
+jest.mock('../../../../../../modules/datasource/npm');
+jest.mock('../../../../../../util/host-rules');
+
+const hostRules = mocked(_hostRules);
+
+const changelogMd = Fixtures.get('gitter-webapp.md', '../..');
+
+const upgrade = partial<BranchUpgradeConfig>({
+  manager: 'some-manager',
+  branchName: '',
+  endpoint: 'https://api.bitbucket.org/',
+  packageName: 'some-repo',
+  versioning: semverVersioning.id,
+  currentVersion: '5.2.0',
+  newVersion: '5.7.0',
+  sourceUrl: 'https://bitbucket.org/some-org/some-repo',
+  releases: [
+    { version: '5.2.0' },
+    {
+      version: '5.4.0',
+      releaseTimestamp: '2018-08-24T14:23:00.000Z',
+    },
+    { version: '5.5.0', gitRef: 'abcd' },
+    {
+      version: '5.6.0',
+      gitRef: '1234',
+      releaseTimestamp: '2020-02-13T15:37:00.000Z',
+    },
+    { version: '5.6.1', gitRef: 'asdf' },
+  ],
+});
+
+const bitbucketTreeResponse = {
+  values: [
+    {
+      type: 'commit_directory',
+      path: 'lib',
+      commit: {
+        hash: '1234',
+      },
+    },
+    {
+      type: 'commit_file',
+      path: 'CHANGELOG.md',
+      commit: {
+        hash: 'abcd',
+      },
+    },
+    {
+      type: 'commit_file',
+      path: 'RELEASE_NOTES.md',
+      commit: {
+        hash: 'asdf',
+      },
+    },
+  ],
+};
+
+const bitbucketTreeResponseNoChangelogFiles = {
+  values: [
+    {
+      type: 'commit_directory',
+      path: 'lib',
+      commit: {
+        hash: '1234',
+      },
+    },
+  ],
+};
+
+const bitbucketProject = partial<ChangeLogProject>({
+  type: 'bitbucket',
+  apiBaseUrl: 'https://api.bitbucket.org',
+  baseUrl: 'https://bitbucket.org/',
+});
+
+describe('workers/repository/update/pr/changelog/bitbucket/index', () => {
+  beforeEach(() => {
+    hostRules.clear();
+  });
+
+  it('retrieves changelog json', async () => {
+    expect(
+      await getChangeLogJSON({
+        ...upgrade,
+      })
+    ).toEqual({
+      hasReleaseNotes: true,
+      project: {
+        apiBaseUrl: 'https://api.bitbucket.org/',
+        baseUrl: 'https://bitbucket.org/',
+        packageName: 'some-repo',
+        repository: 'some-org/some-repo',
+        sourceDirectory: undefined,
+        sourceUrl: 'https://bitbucket.org/some-org/some-repo',
+        type: 'bitbucket',
+      },
+      versions: [
+        expect.objectContaining({ version: '5.6.1' }),
+        expect.objectContaining({ version: '5.6.0' }),
+        expect.objectContaining({ version: '5.5.0' }),
+        expect.objectContaining({ version: '5.4.0' }),
+      ],
+    });
+  });
+
+  it('generates release notes', async () => {
+    hostRules.find.mockReturnValue({ token: 'some-token' });
+    jest.setTimeout(0);
```

Comment:
```suggestion
```
this can cause hung tests, why was it added?

Reply:
Copied it from the github and gitlab test suites.  I've removed it